prime-router/src/test/kotlin/azure/ActionHistoryTests.kt

```diff
@@ -908,4 +914,135 @@ class ActionHistoryTests {
             )
         )
     }
+
+    @Test
+    fun `test queueFhirMessages properly add a fhir messages to be tracked`() {
+        val actionHistory = ActionHistory(TaskAction.receive)
+
+        val queueMessage = FhirConvertQueueMessage(
+            UUID.randomUUID(),
+            "",
+            "",
+            "",
+            Topic.FULL_ELR,
+            ""
+        )
+
+        actionHistory.trackFhirMessage(queueMessage)
+        assertEquals(1, actionHistory.fhirQueueMessages.size)
+    }
+
+    @Test
+    fun `test queueFhirMessages properly adds multiple fhir messages to be tracked`() {
+        val actionHistory = ActionHistory(TaskAction.receive)
+
+        actionHistory.trackFhirMessage(
+            FhirConvertQueueMessage(
+            UUID.randomUUID(),
+            "",
+            "",
+            "",
+            Topic.FULL_ELR,
+            ""
+        )
+        )
+
+        actionHistory.trackFhirMessage(
+            FhirReceiverEnrichmentQueueMessage(
+            UUID.randomUUID(),
+            "",
+            "",
+            "",
+            Topic.FULL_ELR,
+            ""
+        )
+        )
+
+        assertEquals(2, actionHistory.fhirQueueMessages.size)
+    }
+
+    @Test
+    fun `test queueFhirMessages properly tracks and queues a fhir messages`() {
```

Comment:
Good job on these tests! `nit:` you could probably remove the two tests that test adding only a single message, I think they are redundant (if you can add two, why wouldn't you be able to add one?)